strax/context.py

```diff
@@ -1014,6 +965,50 @@ def get_iter(self, run_id: str,
             raise ValueError(f"Invalid time range: {time_range}, "
                              "returned no chunks!")
 
+    def _make_progress_bar(self, run_id, targets, progress_bar=True):
+        """
+        Make a progress bar for get_iter
+        :param run_id, targets: run_id and targets
+        :param progress_bar: Bool weither or not to display the progress bar
+        :return: progress bar, t_start (run) and t_end (run)
+        """
+        try:
+            t_start, t_end, = self.estimate_run_start_and_end(run_id, targets)
+        except (AttributeError, KeyError, IndexError):
+            # During testing some thing remain a secret
+            t_start, t_end, = 0, float('inf')
+        if t_end == float('inf'):
+            progress_bar = False
+
+        # Define nice progressbar format:
+        bar_format = ("{desc}: "  # The loading plugin x
+                      "|{bar}| "  # Bar that is being filled
+                      "{percentage:.2f} % "  # Percentage
+                      "[{elapsed}<{remaining}]"  # Time estimate
+                      "{postfix}"  # Extra info
+                      )
+        pbar = tqdm(total=1,
+                    desc=f'Loading {targets}',
+                    bar_format=bar_format,
+                    leave=True,
+                    disable=not progress_bar)
+        return pbar, t_start, t_end
+
+    @staticmethod
+    def _update_progress_bar(pbar, t_start, t_end, n_chunks,  chunk_end):
+        """Do some tqdm voodoo to get the progress bar for st.get_iter"""
+        if t_end - t_start > 0:
+            pbar.n += (chunk_end - t_start) / (t_end - t_start)/n_chunks
```

Comment:
Is there a particular reason why you changed `pbar.n = (chunk_end - t_start) / (t_end - t_start)`  into `pbar.n += (chunk_end - t_start) / (t_end - t_start)/n_chunks`? For me both is fine, I am just curious.

Reply:
Uhm, now that I look at it, I don't think there is a particular reason. I unfixed things, then noticed I was again (uhg) using a pip installed strax version. Then I noticed I made a mess and had to refix it. So in the end, I think there is no reason.